app/connectors/FinancialInstitutionsConnector.scala

```diff
@@ -47,26 +44,15 @@ class FinancialInstitutionsConnector @Inject() (val config: FrontendAppConfig, v
       .get(url"${config.fIManagementUrl}/crs-fatca-fi-management/financial-institutions/$subscriptionId/$fiId")
       .execute[HttpResponse]
 
-  def addFi(fiDetails: CreateFIDetails)(implicit
+  def addOrUpdateFI(fiDetails: CreateFIDetails, requestType: RequestType)(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
-  ): Future[Either[ErrorDetails, HttpResponse]] =
-    httpClient
-      .post(url"${config.fIManagementUrl}/crs-fatca-fi-management/financial-institutions/create")
-      .withBody(Json.toJson(fiDetails))
+  ): Future[HttpResponse] =
+    (requestType match {
+      case CREATE => httpClient.post(url"${config.fIManagementUrl}/crs-fatca-fi-management/financial-institutions/create")
+      case UPDATE => httpClient.put(url"${config.fIManagementUrl}/crs-fatca-fi-management/financial-institutions/update")
+    }).withBody(Json.toJson(fiDetails))
       .execute[HttpResponse]
-      .map {
-        response =>
-          response.status match {
-            case OK => Right(response)
-            case _  => Left((response.json \ "ErrorDetails").as[ErrorDetails])
-          }
-      }
-      .recoverWith {
-        case e: Exception =>
```

Comment:
Do we not still need a logging message as per AC 3?

Reply:
added the error log